enzyme/Enzyme/GradientUtils.h

```diff
@@ -2098,7 +2098,7 @@ class DiffeGradientUtils : public GradientUtils {
         idx2.push_back(v);
         auto selects = addToDiffe(
             val, BuilderM.CreateExtractValue(dif, ArrayRef<unsigned>(i)),
-            BuilderM, nullptr, idx2);
+            BuilderM, addingType, idx2);
```

Comment:
Save for later? This seems like the wrong way to handle (and worth explicitly testing)